internal/jimmhttp/rebac_admin/identities.go

```diff
@@ -120,17 +120,27 @@ func (s *identitiesService) GetIdentityGroups(ctx context.Context, identityId st
 		Relation:     ofganames.MemberRelation.String(),
 		TargetObject: openfga.GroupType.String(),
 	}, int32(filter.Limit()), filter.Token()) // #nosec G115 accept integer conversion
-
 	if err != nil {
 		return nil, err
 	}
-	groups := make([]resources.Group, len(tuples))
-	for i, t := range tuples {
-		groups[i] = resources.Group{
-			Id:   &t.Target.ID,
-			Name: t.Target.ID,
+
+	groups := make([]resources.Group, 0, len(tuples))
+	for _, t := range tuples {
+		dbGroup, err := s.jimm.GetGroupByUUID(ctx, user, t.Target.ID)
+		if err != nil {
+			// Handle the case where the group was removed from the DB but a lingering OpenFGA tuple still exists.
+			// Don't return an error as that would prevent a user from viewing their groups, instead show the group as "removed".
+			if errors.ErrorCode(err) == errors.CodeNotFound {
```

Comment:
Should we be leaving lingering tuples? Or would .GetGroupsByUUIDs be better in the JIMM layer with tuple cleanup? (Really not sure here just opening up for discussion)

Reply:
We shouldn't be leaving lingering tuples, but that's a separate issue. This is just handling the case that, for some reason, there is a leftover tuple but no group in the database.

Reply:
Not sure what you mean here but perhaps explain in tomorrows standup or offline in element?